docs/src/creating-packages.md

```diff
@@ -0,0 +1,189 @@
+# **5.** Creating Packages
+
+A package is a project with a `name`, `uuid` and `version` entry in the `Project.toml` file `src/PackageName.jl` file that defines the module `PackageName`.
+This file is executed when the package is loaded.
+
+### Generating files for a package
+
+To generate files for a new package, use `pkg> generate`.
+
+```
+(v1.0) pkg> generate HelloWorld
+```
+
+This creates a new project `HelloWorld` with the following files (visualized with the external [`tree` command](https://linux.die.net/man/1/tree)):
+
+```jl
+shell> cd HelloWorld
+
+shell> tree .
+.
+├── Project.toml
+└── src
+    └── HelloWorld.jl
+
+1 directory, 2 files
+```
+
+The `Project.toml` file contains the name of the package, its unique UUID, its version, the author and eventual dependencies:
+
+```toml
+name = "HelloWorld"
+uuid = "b4cd1eb8-1e24-11e8-3319-93036a3eb9f3"
+version = "0.1.0"
+author = ["Some One <someone@email.com>"]
+
+[deps]
+```
+
+The content of `src/HelloWorld.jl` is:
+
+```jl
+module HelloWorld
+
+greet() = print("Hello World!")
+
+end # module
+```
+
+We can now activate the project and load the package:
+
+```jl
+pkg> activate .
+
+julia> import HelloWorld
+
+julia> HelloWorld.greet()
+Hello World!
+```
+
+### Adding dependencies to the project
+
+Let’s say we want to use the standard library package `Random` and the registered package `JSON` in our project.
+We simply `add` these packages (note how the prompt now shows the name of the newly generated project,
+since we are inside the `HelloWorld` project directory):
+
+```
+(HelloWorld) pkg> add Random JSON
+ Resolving package versions...
+  Updating "~/Documents/HelloWorld/Project.toml"
+ [682c06a0] + JSON v0.17.1
+ [9a3f8284] + Random
+  Updating "~/Documents/HelloWorld/Manifest.toml"
+ [34da2185] + Compat v0.57.0
+ [682c06a0] + JSON v0.17.1
+ [4d1e1d77] + Nullables v0.0.4
+ ...
+```
+
+Both `Random` and `JSON` got added to the project’s `Project.toml` file, and the resulting dependencies got added to the `Manifest.toml` file.
+The resolver has installed each package with the highest possible version, while still respecting the compatibility that each package enforce on its dependencies.
+
+We can now use both `Random` and `JSON` in our project. Changing `src/HelloWorld.jl` to
+
+```
+module HelloWorld
+
+import Random
+import JSON
+
+greet() = print("Hello World!")
+greet_alien() = print("Hello ", Random.randstring(8))
+
+end # module
+```
+
+and reloading the package, the new `greet_alien` function that uses `Random` can be used:
+
+```
+julia> HelloWorld.greet_alien()
+Hello aT157rHV
+```
+
+### Adding a build step to the package.
+
+The build step is executed the first time a package is installed or when explicitly invoked with `build`.
+A package is built by executing the file `deps/build.jl`.
+
+```
+shell> cat deps/build.log
```

Comment:
AFAICT `deps` doesn't even exist as a directory at this point, so for anyone following along it will fail. And one should illustrate by creating `build.jl`?